frontend/src/components/RoomList/RoomList.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+    Table,
+    TableContainer,
+    TableRow,
+    TableHead,
+    TableCell,
+    makeStyles,
+    TableBody,
+    Button,
+    IconButton
+} from "@material-ui/core";
+import { Add } from "@material-ui/icons";
+import { Alert } from "@material-ui/lab";
+import React, { useContext, useEffect, useState } from "react";
+import { useHistory } from "react-router";
+
+import { Context } from "../../context/appContext";
+import { Room } from "../../model/room";
+import { apiProvider } from "../../services/api";
+import { Navbar } from "../Navbar/Navbar";
+import { NewRoomDialog } from "./NewRoomDialog";
+
+const useStyles = makeStyles({
+    tableHead: {
+        backgroundColor: "#2a9df4"
+    },
+    table: {
+        width: "80%",
+        marginLeft: "10%",
+        marginTop: 100
+    },
+    joinInButton: {
+        width: "100%",
+        color: "white",
+        backgroundColor: "#2a9df4",
+        "&:hover, &:focus": {
+            backgroundColor: "#187bcd"
+        }
+    },
+    countCell: {
+        paddingLeft: 35
+    },
+    error: {
+        width: "78.5%",
+        marginLeft: "10%",
+        marginTop: 10
+    }
+});
+
+export function RoomList() {
+    const [{ socket }, dispatch] = useContext(Context);
+    const [rooms, setRooms] = useState<Room[]>([]);
+    const [openDialog, setOpenDialog] = useState(false);
+    const [apiErrorMessage, setAprErrorMessage] = useState<string>("");
+    const history = useHistory();
+
+    useEffect(() => {
+        socket.on("newRoom", (room: Room) => {
+            setRooms([...rooms, room]);
+        });
+    }, [rooms, socket]);
+
+    useEffect(() => {
+        const getRooms = async () => {
```

Comment:
you must be kidding me... Do you even know why you use this or just copy/paste from sneaker chain?